sleap_roots/graphpipeline.py

```diff
@@ -169,9 +188,287 @@ def get_traits_value_frame(
+            long_primary_pts = primary_pts[max_length_idx]
+            primary_pts = np.reshape(
+                long_primary_pts,
+                (1, long_primary_pts.shape[0], long_primary_pts.shape[1]),
+            )
+        else:
+            # if no primary root, just give two nan points
+            primary_pts = np.array([[(np.nan, np.nan), (np.nan, np.nan)]])
+            pts_all_array = lateral_pts
+            pts_all_list = list(lateral_pts)
+        # if no lateral root, just give two nan points
+        if get_root_lengths(lateral_pts).shape[0] == 0:
+            lateral_pts = np.array([[(np.nan, np.nan), (np.nan, np.nan)]])
+
+        data = get_traits_value_frame(
+            primary_pts, lateral_pts, pts_all_array, pts_all_list
+        )
+        data["plant_name"] = plant_name
+        data["frame_idx"] = frame
+        data_plant = pd.concat([data_plant, data], ignore_index=True)
+
+    # reorganize the column position
+    column_names = data_plant.columns.tolist()
+    column_names = [column_names[-2]] + [column_names[-1]] + column_names[:-2]
+    data_plant = data_plant[column_names]
+
+    if write_csv:
+        data_plant.to_csv("plant_original_traits.csv", index=False)
+    return data_plant
+
+
+def get_traits_value_plant_summary(
+    h5,
+    lateral_only: bool = False,
+    write_csv: bool = False,
+) -> pd.DataFrame:
+    """Get summarized SLEAP traits per plant based on graph.
+
+    Args:
+        h5: h5 file, plant image series.
+        lateral_only: Boolean value, where false is rice (default), true is dicot.
+        write_csv: Boolean value, where true is write csv file.
+
+    Return:
+        A DataFrame with all summarized traits per plant.
+    """
+    # lateral_only = False
+    # write_csv = False
+    data_plant = get_traits_value_plant(h5, lateral_only, write_csv)
+
+    scalar_traits = [
+        "primary_angle_proximal",
+        "primary_angle_distal",
+        "primary_length",
+        "primary_base_tip_dist",
+        "primary_depth",
+        "lateral_count",
+        "grav_index",
+        "base_length",
+        "base_length_ratio",
+        "primary_tip_pt_y",
+        "base_median_ratio",
+        "base_ct_density",
+        "chull_perimeter",
+        "chull_area",
+        "chull_max_width",
+        "chull_max_height",
+        "ellipse_a",
+        "ellipse_b",
+        "ellipse_ratio",
+        "network_width_depth_ratio",
+        "network_solidity",
+        "network_length_lower",
+        "network_distribution_ratio",
+        "scanline_first_ind",
+        "scanline_last_ind",
+    ]
+
+    non_scalar_traits = [
+        "lateral_angles_proximal",
+        "lateral_angles_distal",
+        "lateral_lengths",
+        "stem_widths",
+        "lateral_base_xs",
+        "lateral_base_ys",
+        "lateral_tip_xs",
+        "lateral_tip_ys",
+        "chull_line_lengths",
+        "scanline_intersection_counts",
+    ]
```

Comment:
Make these module-level constants (after the imports but before functions) and tuples. Like:

```py
SCALAR_TRAITS = (
    "primary_angle_proximal",
    "primary_angle_distal",
    "primary_length",
    "primary_base_tip_dist",
    "primary_depth",
    "lateral_count",
    "grav_index",
    "base_length",
    "base_length_ratio",
    "primary_tip_pt_y",
    "base_median_ratio",
    "base_ct_density",
    "chull_perimeter",
    "chull_area",
    "chull_max_width",
    "chull_max_height",
    "ellipse_a",
    "ellipse_b",
    "ellipse_ratio",
    "network_width_depth_ratio",
    "network_solidity",
    "network_length_lower",
    "network_distribution_ratio",
    "scanline_first_ind",
    "scanline_last_ind",
)

NON_SCALAR_TRAITS = (
    "lateral_angles_proximal",
    "lateral_angles_distal",
    "lateral_lengths",
    "stem_widths",
    "lateral_base_xs",
    "lateral_base_ys",
    "lateral_tip_xs",
    "lateral_tip_ys",
    "chull_line_lengths",
    "scanline_intersection_counts",
)
```

This makes these constants accessible from elsewhere and provides a single point of reference where these trait names are defined.